src/UnigramWordPredictor.java

```diff
@@ -50,10 +51,20 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
    */
   public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
-
+    Map<String, List<String>> neighborMap = new HashMap<>();
     // TODO: Convert the trainingWords into neighborMap here
+    for (int i = 0; i < trainingWords.size() - 1; i++) {
+      String current = trainingWords.get(i);
+      String next = trainingWords.get(i + 1); 
+      if (!neighborMap.containsKey(current)){
+        neighborMap.put(current, new ArrayList<>());
+      }
+      neighborMap.get(current).add(next);
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?